homeassistant/components/enasolar/config_flow.py

```diff
@@ -0,0 +1,223 @@
+"""EnaSolar solar inverter configuration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import pyenasolar
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    CONF_CAPABILITY,
+    CONF_DC_STRINGS,
+    CONF_HOST,
+    CONF_MAX_OUTPUT,
+    CONF_NAME,
+    CONF_SUN_DOWN,
+    CONF_SUN_UP,
+    DEFAULT_HOST,
+    DEFAULT_NAME,
+    DEFAULT_SUN_DOWN,
+    DEFAULT_SUN_UP,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def enasolar_entries(hass: HomeAssistant):
+    """Return the hosts already configured."""
+    return {
+        entry.data[CONF_HOST] for entry in hass.config_entries.async_entries(DOMAIN)
+    }
+
+
+class EnaSolarConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """User Configuration of EnaSolar Integration."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._errors: dict[str, str] = {}
+        self._enasolar = None
+        self._data: dict[str, Any] = {}
+
+    def _host_in_configuration_exists(self, host) -> bool:
+        """Return True if host exists in configuration."""
+        if host in enasolar_entries(self.hass):
+            return True
+        return False
+
+    def _get_serial_no(self):
+        return self._enasolar.get_serial_no()
+
+    def _get_max_output(self):
+        return self._enasolar.get_max_output()
+
+    def _get_dc_strings(self):
+        return self._enasolar.get_dc_strings()
+
+    def _get_capability(self):
+        return self._enasolar.get_capability()
+
+    def _get_name(self):
+        return self._data[CONF_NAME]
+
+    async def _try_connect(self, host):
+        kwargs = {}
+
+        try:
+            self._enasolar = pyenasolar.EnaSolar(host, **kwargs)
+            await self._enasolar.interogate_inverter()
+            return True
+
+        except Exception as e:
+            self._errors[CONF_HOST] = "cannot_connect"
+            _LOGGER.error("Connection to EnaSolar inverter '%s' failed (%s)", host, e)
+        return False
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get options flow for this handler."""
+        return EnaSolarOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Step when user initializes a integration."""
+        self._errors = {}
+
+        if user_input is not None:
+            self._data[CONF_HOST] = user_input[CONF_HOST]
+            if self._host_in_configuration_exists(self._data[CONF_HOST]):
+                self._errors[CONF_HOST] = "already_configured"
+            self._data[CONF_NAME] = user_input[CONF_NAME]
```

Comment:
same comment here about setting each key manually vs updating the dictionary

Reply:
As above